broker/value_added/value_added.py

```diff
@@ -0,0 +1,211 @@
+    """ Creates a list of tuples formatted for RAPID classifier.
+
+    Args:
+        alert_list  (list): list of alert dicts
+        xmatch_list (list): list of cross matched objects as given by xm.get_xmatches()
+        survey       (str): name of survey generating the alerts
+
+    Returns:
+        light_curves (list): list of tuples formatted as required for input to RAPID.
+
+        cx_data      (list): list of dicts of candidate and xmatch info as needed for
+                             upload to BigQuery classification table.
+
+    """
+    # throw an error if received non-ZTF data
+    assert (survey=='ZTF'), "\nvalue_added.format_for_rapid() requires survey=='ZTF'"
+
+    light_curves = []
+    cx_dicts = {} # collect candidate and xmatch data to merge with RAPID results
+    oid_map = map_objectId_list(xmatch_list) # maps objectId's to list positions
+
+    for alert in alert_list:
+        oid = alert['objectId']
+        cid = alert['candidate']['candid']
+        cand_mjd = jd_to_mjd(alert['candidate']['jd'])
+        ra = alert['candidate']['ra']
+        dec = alert['candidate']['dec']
+
+        # Observation epoch data
+        fid_dict = {1:'g', 2:'r', 3:'i'}
+        mjd, flux, fluxerr, passband, photflag = ([] for i in range(5))
+
+        epochs = alert['prv_candidates'] + [alert['candidate']]
+        # if one epoch is missing a zeropoint, they should all be missing
+        zp_fallback, zp_in_keys = 26.0, [0 for i in range(len(epochs))]
+        for n, epoch in enumerate(epochs):
+            try:
+                assert epoch['magpsf'] is not None # magpsf is null for nondetections
+                # test this. try setting magnitude to epoch['diffmaglim'] instead of skipping.
+            except:
+                continue # move to next epoch
+
+            # early schema(s) did not contain a magnitude zeropoint
+            if 'magzpsci' not in epoch.keys(): # fix this. do something better.
+                _warn('Epoch does not have zeropoint data. Setting to {}'.format(zp_fallback))
+                zp_in_keys[n] = 1
+                epoch['magzpsci'] = zp_fallback
+
+            mjd.append(jd_to_mjd(epoch['jd']))
+            f, ferr = mag_to_flux(epoch['magpsf'],epoch['magzpsci'],epoch['sigmapsf'])
+            flux.append(f)
+            fluxerr.append(ferr)
+            passband.append(fid_dict[epoch['fid']])
+            photflag.append(4096)  # fix this, determines trigger time
+                                    # (1st mjd where this == 6144)
+
+        # check that either all or no epochs with detections have missing zeropoint
+        if sum(zp_in_keys) not in [0, len(mjd)]:
+            err_msg = ("Inconsistent zeropoint values in the epochs of alert {}."
+                        "Cannot continue with classification.").format(oid)
+            assert False, err_msg
+
+        # Set trigger date. fix this.
+        photflag[np.where(flux==np.max(flux))[0][0]] = 6144
```

Comment:
I find `photflag[np.equals(flux == np.max(flux))] = 6144` slightly more readable, but that's personal preference.

Reply:
I like it, thanks

Reply:
Actually `photflag[np.equals(flux == np.max(flux))] = 6144` fails with `AttributeError: module 'numpy' has no attribute 'equals'` and I can't find anything online indicating that the function exists. I thought maybe you meant `np.equal`, but that requires a different syntax and fails with `TypeError: only integer scalar arrays can be converted to a scalar index`.

What did you mean here?